controllers/machinedeployment_controller.go

```diff
@@ -274,10 +274,11 @@ func (r *MachineDeploymentReconciler) getMachineSetsForDeployment(ctx context.Co
 		// Attempt to adopt machine if it meets previous conditions and it has no controller references.
 		if metav1.GetControllerOf(ms) == nil {
 			if err := r.adoptOrphan(ctx, d, ms); err != nil {
-				r.recorder.Eventf(d, corev1.EventTypeWarning, "FailedAdopt", "Failed to adopt MachineSet %q: %v", ms.Name, err)
 				log.Error(err, "Failed to adopt MachineSet into MachineDeployment", "machineset", ms.Name)
+				r.recorder.Eventf(d, corev1.EventTypeWarning, "FailedAdopt", "Failed to adopt MachineSet %q: %v", ms.Name, err)
 				continue
 			}
+			log.Error(err, "Adopted MachineSet into MachineDeployment", "machineset", ms.Name)
```

Comment:
log.Error? also, to double check, MD name is already in the logger as a reconcile object?

Reply:
Thx for catching that, copy&paste :/

Yes the MD is already there: (added by controller-runtime as it's the reconciled object), e.g.
```bash
I1103 10:07:41.600804       1 machinedeployment_controller.go:123] controller/machinedeployment "msg"="Reconciliation is paused for this object" "name"="quick-start-vg4pi9-md-0" "namespace"="quick-start-s8xx5t" "reconciler group"="cluster.x-k8s.io" "reconciler kind"="MachineDeployment" 
```